IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c

```diff
@@ -0,0 +1,83 @@
+/** @file
+  IPMI Command - NetFnSensor.
+
+  Copyright (c) Microsoft Corporation
+  PDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/IpmiBaseLib.h>
+
+#include <IpmiFeature.h>
+
+/**
+ * @brief Send the Set Sensor Threshold command for a specified SensorNumber
+ *
+ * @param SetSensorThresholdRequestData The filled out Set Sensor Threshold command structure
+ * @param CompletionCode Pointer to a buffer for returning the completion code
+ * @return EFI_STATUS
+ */
+EFI_STATUS
+EFIAPI
+IpmiSetSensorThreshold (
+  IN IPMI_SENSOR_SET_SENSOR_THRESHOLD_REQUEST_DATA  *SetSensorThresholdRequestData,
+  OUT UINT8                                         *CompletionCode
+  )
+{
+  EFI_STATUS  Status;
+  UINT32      ResponseDataSize;
+
+  Status = EFI_INVALID_PARAMETER;
+
+  if ((SetSensorThresholdRequestData != NULL) && (CompletionCode != NULL)) {
+    ResponseDataSize = sizeof (*CompletionCode);
+
+    Status = IpmiSubmitCommand (
+               IPMI_NETFN_SENSOR_EVENT,
+               IPMI_SENSOR_SET_SENSOR_THRESHOLDS,
+               (UINT8 *)SetSensorThresholdRequestData,
+               sizeof (*SetSensorThresholdRequestData),
+               (UINT8 *)&CompletionCode,
+               &ResponseDataSize
+               );

```

Comment:
My understanding is that `IpmiSubmitCommand()` takes a `UINT8*` for the 5th parameter (`ResponseData`):

```c
EFI_STATUS
IpmiSubmitCommand (
  IN UINT8       NetFunction,
  IN UINT8       Command,
  IN UINT8       *CommandData,
  IN UINT32      CommandDataSize,
  OUT UINT8      *ResponseData,
  IN OUT UINT32  *ResponseDataSize
  );
```

Wouldn't it just write to the `CompletionCode` buffer here?

```suggestion
    Status = IpmiSubmitCommand (
               IPMI_NETFN_SENSOR_EVENT,
               IPMI_SENSOR_SET_SENSOR_THRESHOLDS,
               (UINT8 *)SetSensorThresholdRequestData,
               sizeof (*SetSensorThresholdRequestData),
               CompletionCode,
               &ResponseDataSize
               );

```